src/util/config/index.ts

```diff
@@ -0,0 +1,64 @@
+import { getEnv, intoNumber, intoBoolean } from './util';
+
+export enum Environment {
+	Dev = 'dev',
+	Production = 'production'
+}
+
+interface EnvConfig {
+	port: number;
+	environment: Environment;
+	useSSL: boolean;
+	db: {
+		type: string;
+		host: string;
+		port: number;
+		user: string;
+		password: string;
+		database: string;
+	};
+	hs: {
+		applicationUrl: string;
+		authUrl: string;
+	};
+	sessionSecret: string;
+	googleAnalyticsId: string;
+	dropboxToken: string;
+	sendgridToken: string;
+}
+
+export function load(source: Record<string, string | undefined> = process.env): EnvConfig {
```

Comment:
Everytime we call the `getConfig` function it calls this `load` function. I think it may be good if we create this object once at startup and we return the cached version of the object whenever the `getConfig` function is called to reduce the overhead a bit.

Reply:
It should only load() the config if the refresh parameter is true or if the config has never been loaded, once it's been loaded it's stored in `globalConfig` and next time getConfig() is called it should just return that again.